docs/signature-layout.md

```diff
@@ -0,0 +1,35 @@
+# Signature File Layout
+
+A common file layout for storing and serving signatures provides a consistent way to reference image signatures. Signatures on a filesystem or a web server shall use this common layout. Signatures stored in a REST API are not required to use this common layout.
+
+## Specification
+
+This specification relies on [RFC3986](https://tools.ietf.org/html/rfc3986), focusing on defining a [path component](https://tools.ietf.org/html/rfc3986#section-3.3) to compose a concise URI reference to a signature.
+
+**SCHEME[AUTHORITY]/PATH_PREFIX/IMAGE@MANIFEST_DIGEST/signature-INT**
+
+**Definitions**
+
+* **SCHEME**: URI scheme per [RFC3986](https://tools.ietf.org/html/rfc3986#section-3.1), e.g. **file://** or **https://**
+* **AUTHORITY**: An optional authority reference per [RFC3986](https://tools.ietf.org/html/rfc3986#section-3.2), e.g. **example.com**
+* **PATH_PREFIX**: An arbitrary base path to the image component
+* **IMAGE**: The name of the image per [v2 API](https://docs.docker.com/registry/spec/api/#/overview). This would typically take the form of registry/repository/image but is not required to have exactly three parts. There is no requirement to include a **:PORT** component but it should be included if part of the image reference.
```

Comment:
@mtrmac I definitely see where you are coming from. Attaching signatures to manifests makes the logistics of signature distribution MUCH easier. However, per https://github.com/containers/image/pull/59, we are discussing "a detached container image signature object and signing methods".

If we instead are going to discuss an attached model, that would be entirely different. I like where you're going, thinking about signatures as an additional entity managed by a registry, side-by-side with manifests in the same namespace. But at this point it does not seem realistic to get it added to a registry protocol, or else we should be having this discussion elsewhere.

As such, I don't see how we can talk about this as "an invisible implementation detail simulating real registries with native signature support". How do we simulate something that doesn't exist?

We could consider requiring that a sigstore be tightly coupled to a specific registry. When you deploy a registry, you deploy a sigstore with it whose namespaces will match those in the registry. A problem though is that every tool that could move images around, from one registry to another, must also be aware of accompanying sigstores and move signatures together with images. Now we would need a standard API for adding signatures to a sigstore and copying them within a sigstore. Every tool that manipulates a registry would need to use that API to manipulate the accompanying sigstore.

The problem not solved by the namespaced-sigstore model, as this PR defines it, is one of mirroring. Consider that I have an on-premises registry at ``reg.hrivnak.org``. I use it to have a local mirror of third-part content, so it includes repositories like ``redhat/rhel7``. A local client tries to ``pull reg.hrivnak.org/redhat/rhel7:latest`` and gets some content. At what URL should it expect to find a signature?

The client has no idea, nor any need to know, that the image came from a ``redhat.com`` registry. It cannot use a ``redhat.com`` hostname in constructing a signature URL, because it has no awareness that the content originated there. This is a missed use case in the current proposal.

A flat directory of signatures would easily solve this issue. It has disadvantages, but it's strength is this: because a manifest is not inherently tied to a specific registry, its signatures also should not be tied to a specific registry.

Access control with a flat namespace could be reasonably achieved with multiple sigstores. If you have a secret key that you use to sign things for secret purposes, put those signatures in a secret/protected sigstore. This allows you to orient a sigstore with respect to an authority and an audience, which is primarily how content gets consumed. _As a user, I want to get signatures from a particular authority that I trust. As an authority, I want to make my signatures available to a defined set of users._ Thus the practical relationship between a signing authority and its audience may not need to include registry names or repository namespaces.

@mtrmac I have a feeling that you have a particular scenario in mind that involves many users, many keys, and lots of access restrictions. Perhaps it would help to define those use cases directly and explicitly, and then we could more easily see which designs would accommodate them.

I am not at all tied to the idea of a flat namespace, but I am very interested in covering use cases that involve mirroring.